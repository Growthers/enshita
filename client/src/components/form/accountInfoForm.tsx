import { FC, useState } from "react";
import styles from "~/styles/components/form/accountInfoForm.module.scss";
import { SubmitHandler, useForm } from "react-hook-form";
import { Icon } from '@iconify/react';
import { zodResolver } from "@hookform/resolvers/zod";
import * as z from "zod";
import { client } from "~/libs/apiClient";

type FormDataProps = {
  mail: string;
  userName: string;
  newPassword: string;
  currentPassword: string;
};

type Props = {
  mail: string;
  userName: string;
};

const schema = z.object({
  email: z.string().email(),
  userName: z
    .string()
    .min(3)
    .max(128)
    .regex(/^[\w_]+?$/, {
      message: "Contains invalid characters(able to use [a-z],[A-Z],[0-9],[_])",
    }),
  newPassword: z
    .string()
    .min(3)
    .max(128)
    .regex(/^[\w_-]+?$/, {
      message: "Contains invalid characters(able to use [a-z],[A-Z],[0-9])",
    }),
  currentPassword: z
    .string()
    .min(3)
    .max(128)
    .regex(/^[\w_-]+?$/, {
      message: "Contains invalid characters(able to use [a-z],[A-Z],[0-9])",
    }),
});

const AccountInfoForm: FC<Props> = ({ mail, userName }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<FormDataProps>({
    resolver: zodResolver(schema),
    defaultValues: {
      mail,
      userName,
    },
  });

  const [IsNotUpDate, setNotUpdate] = useState(false);
  const [IsShow, setShow] = useState(true);

  const onSubmit: SubmitHandler<FormDataProps> = data => {
    client
      .post("", {
        mail: data.mail,
        name: data.userName,
        newPassword: data.newPassword,
        currentPassword: data.currentPassword,
      })
      .then(() => {})
      .catch(() => {
        setNotUpdate(true);
      });
  };

  return (
    <div className={styles["enshita-accountinfoform-main"]}>
      <div className={styles["enshita-accountinfoform-head"]}>
        <h1>Personal Info</h1>
      </div>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className={styles["enshita-accountinfoform-form-body"]}>
          <label htmlFor="mail/name">
            <small>mail Address</small>
            <div className={styles["enshita-accountinfoform-textarea"]}>
              <div className={styles["enshita-accountinfoform-icon"]}>
                <Icon icon="ci:mail" width="15" height="15"/>
              </div>
              <input
                className={styles["enshita-accountinfoform-input-i1"]}
                {...register("mail", { required: true })}
              />
            </div>
            <div className={styles["enshita-accountinfoform-error"]}>
              {errors.mail && <small>this field is required</small>}
            </div>
          </label>
        </div>

        <div className={styles["enshita-accountinfoform-form-body"]}>
          <label htmlFor="mail/name">
            <small>Username</small>
            <div className={styles["enshita-accountinfoform-textarea"]}>
              <div className={styles["enshita-accountinfoform-icon"]}>
              <Icon icon="ant-design:user-outlined" width="15" height="15"/>
              </div>
              <input
                className={styles["enshita-accountinfoform-input-i1"]}
                {...register("userName", { required: true })}
              />
            </div>
            <div className={styles["enshita-accountinfoform-error"]}>
              {errors.userName && (
                <p>
                  <small>{`${
                    errors.userName?.ref?.value === ""
                      ? "this field is required"
                      : errors.userName?.message
                  }`}</small>
                </p>
              )}
            </div>
          </label>
        </div>

        <div className={styles["enshita-accountinfoform-dis"]}>
          <small>
            if you update your password, you must enter your current password.
          </small>
        </div>

        <div className={styles["enshita-accountinfoform-form-body"]}>
          <label htmlFor="password">
            <small>New Password</small>
            <div className={styles["enshita-accountinfoform-textarea"]}>
              <div className={styles["enshita-accountinfoform-icon"]}>
                <Icon icon="bxs:lock-alt" width="15" height="15"/>
              </div>
              <input
                type={IsShow ? "password" : "text"}
                className={styles["enshita-accountinfoform-input-i2"]}
                {...register("newPassword", { required: true })}
              />
              <button
                type="button"
                className={styles["enshita-accountinfoform-icon"]}
                onClick={() => {
                  setShow(!IsShow);
                }}
              >
                {IsShow && <Icon icon="ant-design:eye-filled" width="15" height="15"/>}
                {!IsShow && <Icon icon="ant-design:eye-invisible-filled" width="15" height="15"/>}
              </button>
            </div>
            <div className={styles["enshita-accountinfoform-error"]}>
              {errors.newPassword && (
                <p>
                  <small>{`${
                    errors.newPassword.ref?.value === ""
                      ? "this field is required"
                      : errors.newPassword.message
                  }`}</small>
                </p>
              )}
            </div>
          </label>
        </div>

        <div className={styles["enshita-accountinfoform-form-body"]}>
          <label htmlFor="password">
            <small>Current Password</small>
            <div className={styles["enshita-accountinfoform-textarea"]}>
              <div className={styles["enshita-accountinfoform-icon"]}>
                <Icon icon="bxs:lock-alt" width="15" height="15"/>
              </div>
              <input
                type={IsShow ? "password" : "text"}
                className={styles["enshita-accountinfoform-input-i2"]}
                {...register("currentPassword", { required: true })}
              />
              <button
                type="button"
                className={styles["enshita-accountinfoform-icon"]}
                onClick={() => {
                  setShow(!IsShow);
                }}
              >
                {IsShow && <Icon icon="ant-design:eye-filled" width="15" height="15"/>}
                {!IsShow && <Icon icon="ant-design:eye-invisible-filled" width="15" height="15"/>}
              </button>
            </div>
            <div className={styles["enshita-accountinfoform-error"]}>
              {errors.currentPassword && (
                <p>
                  <small>{`${
                    errors.currentPassword.ref?.value === ""
                      ? "this field is required"
                      : errors.currentPassword.message
                  }`}</small>
                </p>
              )}
              {IsNotUpDate &&
                !errors.currentPassword &&
                !errors.mail &&
                !errors.userName && (
                  <p>
                    <small>failed update</small>
                  </p>
                )}
            </div>
          </label>
        </div>
        <div className={styles["enshita-accountinfoform-button"]}>
          <button type="submit" onClick={() => setNotUpdate(false)}>
            update
          </button>
        </div>
      </form>
    </div>
  );
};

export default AccountInfoForm;
