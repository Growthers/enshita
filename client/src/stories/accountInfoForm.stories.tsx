// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
