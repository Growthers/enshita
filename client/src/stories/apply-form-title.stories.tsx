import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
