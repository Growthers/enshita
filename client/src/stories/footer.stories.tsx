// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
