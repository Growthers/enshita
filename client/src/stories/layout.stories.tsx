import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
