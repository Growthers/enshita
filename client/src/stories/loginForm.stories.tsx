// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
