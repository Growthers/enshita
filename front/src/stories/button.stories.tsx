import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
