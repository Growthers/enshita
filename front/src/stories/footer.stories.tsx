import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
