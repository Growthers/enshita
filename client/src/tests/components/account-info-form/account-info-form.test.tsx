import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/account-info-form/accountInfoForm.stories";

const { Default } = composeStories(stories);

describe("(component) account information update form test", () => {
  test("Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
