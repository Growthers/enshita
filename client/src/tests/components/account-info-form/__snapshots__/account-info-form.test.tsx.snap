// Vitest Snapshot v1

exports[`(component) account information update form test > Snap Shot 1`] = `
<div>
  <div
    class="_enshita-form-main_uoaol_1"
  >
    <div
      class="_enshita-form-head_uoaol_8"
    >
      <h1>
        Personal Info
      </h1>
    </div>
    <form>
      <section
        class="_enshita-control-body_mclyn_1"
      >
        <label
          for="mail"
        >
          email Address
        </label>
        <div
          class="_enshita-input-textarea_7x1ku_1 undefined"
        >
          <div
            class="_enshita-input-icon_7x1ku_15"
          >
            <span />
          </div>
          <input
            class="_enshita-input-normal_7x1ku_23"
            id="mail"
            name="mail"
            value="abc@example.com"
          />
        </div>
        <p
          class="_enshita-input-error_7x1ku_37"
        />
      </section>
      <section
        class="_enshita-control-body_mclyn_1"
      >
        <label
          for="userName"
        >
          User name
        </label>
        <div
          class="_enshita-input-textarea_7x1ku_1 undefined"
        >
          <div
            class="_enshita-input-icon_7x1ku_15"
          >
            <span />
          </div>
          <input
            class="_enshita-input-normal_7x1ku_23"
            id="userName"
            name="userName"
            value="hoge"
          />
        </div>
        <p
          class="_enshita-input-error_7x1ku_37"
        />
      </section>
      <div
        class="_enshita-form-dis_uoaol_18"
      >
        <small>
          if you update your password, you must enter your current password.
        </small>
      </div>
      <section
        class="_enshita-control-body_mclyn_1"
      >
        <label
          for="newPassword"
        >
          New Password
        </label>
        <div
          class="_enshita-input-textarea_7x1ku_1 undefined"
        >
          <div
            class="_enshita-input-icon_7x1ku_15"
          >
            <span />
          </div>
          <input
            class="_enshita-input-password_7x1ku_30"
            id="newPassword"
            name="newPassword"
            type="password"
            value=""
          />
          <button
            class="_enshita-input-icon_7x1ku_15"
            type="button"
          >
            <span />
          </button>
        </div>
        <p
          class="_enshita-input-error_7x1ku_37"
        />
      </section>
      <section
        class="_enshita-control-body_mclyn_1"
      >
        <label
          for="reNewPassword"
        >
          New Password (Re-enter)
        </label>
        <div
          class="_enshita-input-textarea_7x1ku_1 undefined"
        >
          <div
            class="_enshita-input-icon_7x1ku_15"
          >
            <span />
          </div>
          <input
            class="_enshita-input-password_7x1ku_30"
            id="reNewPassword"
            name="reNewPassword"
            type="password"
            value=""
          />
          <button
            class="_enshita-input-icon_7x1ku_15"
            type="button"
          >
            <span />
          </button>
        </div>
        <p
          class="_enshita-input-error_7x1ku_37"
        />
      </section>
      <section
        class="_enshita-control-body_mclyn_1"
      >
        <label
          for="currentPassword"
        >
          Current Password
        </label>
        <div
          class="_enshita-input-textarea_7x1ku_1 undefined"
        >
          <div
            class="_enshita-input-icon_7x1ku_15"
          >
            <span />
          </div>
          <input
            class="_enshita-input-password_7x1ku_30"
            id="currentPassword"
            name="currentPassword"
            type="password"
            value=""
          />
          <button
            class="_enshita-input-icon_7x1ku_15"
            type="button"
          >
            <span />
          </button>
        </div>
        <p
          class="_enshita-input-error_7x1ku_37"
        />
      </section>
      <div
        class="_enshita-form-error_uoaol_46"
      >
        <small />
      </div>
      <div
        class="_enshita-form-button_uoaol_25"
      >
        <button
          class="_enshita-button-box_1gcpx_17 _enshita-button-update-box_uoaol_34"
          tabindex="0"
          type="submit"
        >
          <p
            class="_enshita-button-text_1gcpx_36 undefined"
          >
            update
          </p>
        </button>
      </div>
    </form>
  </div>
</div>
`;
